test/TestIntegration.java

```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
+    final PutRequest put3 = new PutRequest(table, "brs3", family, "q", "v3");
+    Deferred.group(client.put(put1), client.put(put2),
+            client.put(put3)).join();
+    // Scan the same 3 rows created above twice.
+    for (int i = 0; i < 2; i++) {
+        LOG.info("------------ iteration #" + i);
+        final Scanner scanner = client.newScanner(table);
+        scanner.setStartKey("brs9");
+        scanner.setStopKey("brs0");
+        scanner.setReversed(true);
+        // Callback class to keep scanning recursively.
+        class cb implements Callback<Object, ArrayList<ArrayList<KeyValue>>> {
+            private int n = 4; // Asserts will start at the last added values
+            public Object call(final ArrayList<ArrayList<KeyValue>> rows) {
+                if (rows == null) {
+                    return null;
+                }
+                n--;
+                try {
+                    assertSizeIs(1, rows);
+                    final ArrayList<KeyValue> kvs = rows.get(0);
+                    final KeyValue kv = kvs.get(0);
+                    assertSizeIs(1, kvs);
+                    assertEq("brs" + n, kv.key());
+                    assertEq("q", kv.qualifier());
+                    assertEq("v" + n, kv.value());
+                    return scanner.nextRows(1).addCallback(this);
+                } catch (AssertionError e) {
+                    // Deferred doesn't catch Errors on purpose, so transform any
+                    // assertion failure into an Exception.
+                    throw new RuntimeException("Asynchronous failure", e);
+                }
+            }
+        }
+        try {
+            // Reversed scanner will return the largest key in the first nextRows call
+            scanner.nextRows(1).addCallback(new cb()).join();
+        } finally {
+            scanner.close().join();
+        }
+    }
+  }
+
+  /** Longer reverse scan, checks that qualifier and value are still in lexico order. */
+  @Test
+  public void reverseLongerScan() throws Exception {
+    client.setFlushInterval(FAST_FLUSH);
+
+    // All puts below should be read in scan
+    final PutRequest put1 = new PutRequest(table, "rl1", family, "q1", "v1");
+    final PutRequest put2 = new PutRequest(table, "rl1", family, "q2", "v2");
+    final PutRequest put3 = new PutRequest(table, "rl2", family, "q3", "v3");
+    final PutRequest put4 = new PutRequest(table, "rl2", family, "q4", "v4");
+    final PutRequest put5 = new PutRequest(table, "rl3", family, "q5", "v5");
+    final PutRequest put6 = new PutRequest(table, "rl3", family, "q6", "v6");
+
+    Deferred.group(Deferred.group(client.put(put1), client.put(put2),
+                  client.put(put3)),
+          Deferred.group(client.put(put4), client.put(put5),
+                  client.put(put6))).join();
+    final Scanner rev_scanner = client.newScanner(table);
+    rev_scanner.setFamily(family);
```

Comment:
Is there a reason we need to set the family in this test?  I noticed you didn't do it in the previous test.  All the rows have the same family, so I assume it's not necessary?